utils/das_data_sorter.py

```diff
@@ -0,0 +1,65 @@
+import pandas as pd 
+import argparse 
+from math import ceil
+
+# accepts terminal commands
+parser = argparse.ArgumentParser()
+parser.add_argument("--file", help="Input CSV file", action="store")
+parser.add_argument("--output", help="Returns the filtered data", default="filtered_data.csv", action="store")
+args = parser.parse_args()
+
+# initialising data
+data = pd.read_csv(args.file)
+time = data["time"]/1000
+gps_course = data["gps_course"]
+gps_speed = data["gps_speed"]
+tempC = data["thermoC"]
+tempF = data["thermoF"]
+cadence = data["cadence"]
+power = data["power"]
+reed_vel = data["reed_velocity"]
+reed_dis = data["reed_distance"]
+
+# sorts time into seconds
+seconds = []
+previous_time = 0
+for i in range(len(time)):
+    if time[i] > previous_time:
+        # sets new previous_time, pushes current millisecond_indices into seconds array, creates new millisecond_indices
+        # millisecond_indices is an array of indexes for milliseconds within the same second
+        if previous_time != 0:
+            #ignores first iteration
+            seconds.append(millisecond_indices)
+
+        previous_time = ceil(time[i])
+        millisecond_indices = []
+
+    # pushes index into current millisecond_indices
+    millisecond_indices.append(i)
+seconds.append(millisecond_indices)
+
+# averages data within the seconds, according to seconds array 
+def avg_Data(array):
```

Comment:
I think it'll be a good exercise for you to write unit tests here in another PR.

Reply:
@khanguslee I've done unit tests using mock data while verifying if an input would have the expected output, which it did. But I've erased these from the script straight after. Do you want me to bring it back?

Reply:
It's been awhile. sorry LOL. Let's work on the unit test after this PR has been merged!